### PR TITLE
Update: add page organized by date

### LINKED ISSUE #117: Idea: link to page with content organized by date of publication
This material is organized by tag, content type and license.

After the last TA5 meeting, I thought if it would it be useful/hard to have a link to the contents organized by publish date, similar to a blog post, so that users could check "what is new".

### PR DESCRIPTION
**Pull request by SeverusYixin**
This PR relate to #117 NFDI4BIOIMAGE/nfdi_search_engine#27
This PR closes #117 NFDI4BIOIMAGE/nfdi_search_engine#27

### CODE REVIEW

**Comment by SeverusYixin** [comment]
Hello @haesleinhuepf, it's me again! Could you take a moment to review this?

I noticed something interesting: your original _toc.yml worked perfectly, but after my update, I had to adjust its position to make it work. Do you have any idea why that might be?

File: docs/_toc.yml [hunk 1]
```diff
@@ -22,11 +22,12 @@ parts:
     chapters:
 {license_toc}
 
-    # when adding new chapters here, don't forget to also add them to the scipts/generate_link_lists.py file
```
Comment (haesleinhuepf) [inline docs/_toc.yml:25]: Why are you removing this?

File: docs/_toc.yml [hunk 2]
```diff
@@ -22,11 +22,12 @@ parts:
...
+    chapters:
+{date_toc}
 
-    
   - caption: Appendix
     chapters:
-    - file: statistics/readme
-    - file: export/readme
-    - file: imprint
+      - file: statistics/readme
```
Comment (haesleinhuepf) [inline docs/_toc.yml:31]: Why are you adding spaces here?

**Review by haesleinhuepf — COMMENTED** [review]
Looks good @SeverusYixin ! Just two minor questions. In general: try to change only things that are relevant for your work and leave the rest untouched

File: docs/_toc.yml (hunk 1 above)
Comment (SeverusYixin) [reply docs/_toc.yml:25]: I mistakenly deleted it when I was editing the script, but I then restored it, but it wasn't updated

File: docs/_toc.yml (hunk 2 above)
Comment (SeverusYixin) [reply docs/_toc.yml:31]: Because without space, the checks cannot be passed, just like other space

Comment (SeverusYixin) [reply docs/_toc.yml:33]: At first, I used your original script and just add "  - caption: What's New
    chapters:
{date_toc}
", but the check can not be passed as the https://github.com/NFDI4BIOIMAGE/training/pull/326/commits/78059c5ccf362a3fefbf1cec29a43c99912f9048. That's why  I asked you as "https://github.com/NFDI4BIOIMAGE/training/pull/326#issuecomment-2437680716"

File: scripts/generate_link_lists.py [hunk 3]
```diff
@@ -63,6 +87,11 @@ def main():
             author_toc += "    - file: " + filename + "\n"    
     replace_in_file(toc_file, "{author_toc}", author_toc)
 
+    # Generate the 'What's New' page with content sorted by publish date
```
Comment (haesleinhuepf) [inline scripts/generate_link_lists.py:90]:
```suggestion
    # Generate the 'What's New' page with content sorted by publication date
```

File: scripts/generate_link_lists.py [hunk 4]
```diff
@@ -63,6 +87,11 @@ def main():
             author_toc += "    - file: " + filename + "\n"    
     replace_in_file(toc_file, "{author_toc}", author_toc)
 
+    # Generate the 'What's New' page with content sorted by publication date
+    date_toc = "    - file: whats_new"  
+    generate_by_publish_date(content, "docs/whats_new.md")
```
Comment (haesleinhuepf) [inline scripts/generate_link_lists.py:92]:
```suggestion
    generate_by_publication_date(content, "docs/whats_new.md")  
```

File: scripts/generate_link_lists.py [hunk 5]
```diff
@@ -1,3 +1,27 @@
+def generate_by_publish_date(content, output_file):
```
Comment (haesleinhuepf) [inline scripts/generate_link_lists.py:1]:
```suggestion
def generate_by_publication_date(content, output_file):
```

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi @SeverusYixin ,

that looks already quite good. I just tested it by building the page and must note that the links do not look nice:

image

I think it would be better if the two links are on two lines and the [Link to content] stuff should dissapper. After modifying the code, consider trying it out and have a look at the page. All you need to execute locally are [these lines](https://github.com/NFDI4BIOIMAGE/training/blob/main/.github/workflows/book.yml#L35-L37).  

Thanks!

Best,
Robert

**Comment by haesleinhuepf** [comment]
Also a hint: If you want this pull-request to close multiple issues, you must write "closes #" multiple times. You can see which issues are closed by hovering with your mouse overt the word "closes":

image

**Comment by SeverusYixin** [comment]
> Also a hint: If you want this pull-request to close multiple issues, you must write "closes #" multiple times. You can see which issues are closed by hovering with your mouse overt the word "closes":
> 
> image

Thank you

**Comment by SeverusYixin** [comment]
> Hi @SeverusYixin ,
> 
> that looks already quite good. I just tested it by building the page and must note that the links do not look nice:
> 
> image
> 
> I think it would be better if the two links are on two lines and the [Link to content] stuff should dissapper. After modifying the code, consider trying it out and have a look at the page. All you need to execute locally are [these lines](https://github.com/NFDI4BIOIMAGE/training/blob/main/.github/workflows/book.yml#L35-L37).
> 
> Thanks!
> 
> Best, Robert

Hi @haesleinhuepf, would you like to merge this PR first, and then create a new issue assigned to me for the Outlook styling improvements?

**Comment by haesleinhuepf** [comment]
I prefer fixing bugs in new pages before they become part of the public website. Thanks!

**Comment by haesleinhuepf** [comment]
Hi @SeverusYixin,

I was just wondering why the lists on this page look different than the lists on other pages? Do you think it would be possible to reuse [code building the pages](https://github.com/NFDI4BIOIMAGE/training/blob/2d2e9e85c541ba4167b798f990f43b9ca9ef85fd/scripts/generate_link_lists.py#L382) also to avoid the bug discussed above?

Thanks!
Robert

**Comment by SeverusYixin** [comment]
> Hi @SeverusYixin,
> 
> I was just wondering why the lists on this page look different than the lists on other pages? Do you think it would be possible to reuse [code building the pages](https://github.com/NFDI4BIOIMAGE/training/blob/2d2e9e85c541ba4167b798f990f43b9ca9ef85fd/scripts/generate_link_lists.py#L382) also to avoid the bug discussed above?
> 
> Thanks! Robert

Let me try it later and give you an answer

**Review by SeverusYixin — COMMENTED** [review]
This is my PR, and I cannot be a reviewer to approve your changes.

**Pull request merged**